packages/aws-cdk-lib/core/lib/private/refs.ts

```diff
@@ -260,7 +269,7 @@ function generateExportName(importStack: Stack, reference: Reference, id: string
   return prefix + localPart.slice(Math.max(0, localPart.length - maxLength + prefix.length));
 }
 
-export function getExportable(stack: Stack, reference: Reference): Reference {
+export function getExportable(stack: Stack, reference: Reference): Intrinsic {
```

Comment:
i understand that this is a private function but I am concerned about the downgrade from returning a `Reference` to an `Intrinsic`. i want to learn more about why thats something you've deemed necessary.

Reply:
This ultimately changed because `referenceNestedStackValueInParent` changed, see that comment for more details.  TL;DR `Fn.split` returns an Intrinsic (not a `Reference`) when untokenized.